services/catalog/src/simcore_service_catalog/core/background_tasks.py

```diff
@@ -0,0 +1,177 @@
+import asyncio
+import logging
+from asyncio.futures import CancelledError
+from pprint import pformat
+from typing import Dict, List, Optional, Set, Tuple
+
+from aiopg.sa import Engine
+from aiopg.sa.connection import SAConnection
+from fastapi import FastAPI
+from pydantic import ValidationError
+from pydantic.types import PositiveInt
+
+from ..api.dependencies.director import get_director_session
+from ..db.repositories.groups import GroupsRepository
+from ..db.repositories.services import ServicesRepository
+from ..models.domain.service import (
+    ServiceAccessRightsAtDB,
+    ServiceDockerData,
+    ServiceMetaDataAtDB,
+)
+
+logger = logging.getLogger(__name__)
+
+ServiceKey = str
+ServiceVersion = str
+
+
+async def _list_registry_services(
+    app: FastAPI,
+) -> Dict[Tuple[ServiceKey, ServiceVersion], ServiceDockerData]:
+    client = get_director_session(app)
+    data = await client.get("/services")
+    services: Dict[Tuple[ServiceKey, ServiceVersion], ServiceDockerData] = {}
+    for x in data:
+        try:
+            service_data = ServiceDockerData.parse_obj(x)
+            services[(service_data.key, service_data.version)] = service_data
+        # services = parse_obj_as(List[ServiceOut], data)
+        except ValidationError as exc:
+            logger.warning(
+                "skip service %s:%s that has invalid fields\n%s",
+                x["key"],
+                x["version"],
+                exc,
+            )
+
+    return services
+
+
+async def _list_db_services(app: FastAPI) -> Set[Tuple[ServiceKey, ServiceVersion]]:
+    engine: Engine = app.state.engine
+    async with engine.acquire() as conn:
+        services_repo = ServicesRepository(conn)
+        return {
+            (service.key, service.version)
+            for service in await services_repo.list_services()
+        }
+
+
+async def _get_everyone_gid(app: FastAPI) -> int:
+    engine: Engine = app.state.engine
+    async with engine.acquire() as conn:
+        groups_repo = GroupsRepository(conn)
+        return (await groups_repo.get_everyone_group()).gid
+
+
+async def _get_service_access_rights(
```

Comment:
For future reference, I would appreciate having a docstring on this function explaining a bit how it works. Like the one int the PR description.

Reply:
good point.